packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -0,0 +1,510 @@
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_set_interrupted_cb failed: %s",
+              get_error_message(ret));
+  }
+
+  ret = player_set_display_visible(player_, true);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_set_display_visible failed: %s",
+              get_error_message(ret));
+  }
+
+  ret = player_set_error_cb(player_, OnError, (void *)this);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_set_error_cb failed: %s",
+              get_error_message(ret));
+  }
+
+  ret = player_prepare_async(player_, OnPrepared, (void *)this);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_prepare_async failed: %s",
+              get_error_message(ret));
+  }
+
+  flutter::PluginRegistrar *plugin_registrar =
+      flutter::PluginRegistrarManager::GetInstance()
+          ->GetRegistrar<flutter::PluginRegistrar>(registrar_ref_);
+  SetupEventChannel(plugin_registrar->messenger());
+  return player_id_;
+}
+
+void VideoPlayer::SetDisplayRoi(int x, int y, int w, int h) {
+  int ret = player_set_display_roi_area(player_, x, y, w, h);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("fail to set display roi :%s", get_error_message(ret));
+  }
+}
+
+VideoPlayer::~VideoPlayer() {
+  if (drm_manager_) {
+    drm_manager_->ReleaseDrmSession();
+  }
+  Dispose();
+}
+
+void VideoPlayer::Play() {
+  player_state_e state;
+  int ret = player_get_state(player_, &state);
+  if (state < PLAYER_STATE_READY) {
+    player_destroy(player_);
+    LOG_ERROR("invalid state for play operation");
+    return;
+  }
+  ret = player_start(player_);
+  if (state == PLAYER_STATE_READY || state == PLAYER_STATE_PAUSED) {
+    if (ret != PLAYER_ERROR_NONE) {
+      player_destroy(player_);
+      LOG_ERROR("fail to start");
+    }
+  }
+}
+
+void VideoPlayer::Pause() {
+  player_state_e state;
+  int ret = player_get_state(player_, &state);
+  if (state <= PLAYER_STATE_READY) {
+    LOG_ERROR("invalid state for pause operation");
+    return;
+  }
+  ret = player_pause(player_);
+  if (state == PLAYER_STATE_PLAYING) {
+    if (ret != PLAYER_ERROR_NONE) {
+      player_destroy(player_);
+      LOG_ERROR("fail to pause");
+    }
+  }
+}
+
+void VideoPlayer::SetLooping(bool is_looping) {
+  LOG_INFO("[VideoPlayer.setLooping] isLooping: %d", is_looping);
+  int ret = player_set_looping(player_, is_looping);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer.setLooping] player_set_looping failed: %s",
+              get_error_message(ret));
+  }
+}
+
+void VideoPlayer::SetVolume(double volume) {
+  LOG_INFO("[VideoPlayer.setVolume] volume: %f", volume);
+  int ret = player_set_volume(player_, volume, volume);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
```

Comment:
If `player_set_volume()` fails for any reason, `player_` is destroyed.
An error may occur in all subsequent actions unless the application is terminated or player_ is re-created.
You need to make sure that calling player_destroy is appropriate.
or you can pass error information to higher level.
Please refer to `video_player_error.h` in the video_player package.

Reply:
> If `player_set_volume()` fails for any reason, `player_` is destroyed. An error may occur in all subsequent actions unless the application is terminated or player_ is re-created. You need to make sure that calling player_destroy is appropriate. or you can pass error information to higher level. Please refer to `video_player_error.h` in the video_player package.

have fixed